File: WEEK1/DAY2/Prog3/P3.py
print("Check if N is a prime number")
N=int(input("Enter N: "))
ans=True
if N<=1:
    ans=False
if N<=3:
    ans=True
if N%2==0 or N%3==0:
    ans=False
i=5
while i*i<=N:
    if N%i==0 or N%(i+2)==0:
        ans=False
        break
    i+=6

if ans==True:
    print("Prime")
else:
    print("Not prime")

File: WEEK1/DAY2/Prog3/P3.c
#include <stdio.h>
int main(){
    printf("Check if N is a prime number\n");
    printf("Enter N: ");
    int N;
    scanf("%d",&N);
    int ans=1;
    if(N<=1){
        ans=0;
    }
    if(N<=3){
        ans=1;
    }
    if(N%2==0 || N%3==0){
        ans=0;
    }
    int i=5;
    while (i*i<=N){
        if (N%i==0 || N%(i+2)==0){
            ans=0;
            break;
        }
        i=i+2;
    }
    if(ans==1){
        printf("Prime\n");
    }
    else{
        printf("Not Prime\n");
    }
}
